### ISSUE TITLE
Running in production

### ISSUE THREAD

**Issue by zegerhoogeboom** [databricks/mlflow#9]
How does one go about running a production server? The startup message when launching the UI warns to not use `mlflow ui` in production:
```
 * Serving Flask app "mlflow.server" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
```

Could you provide an example on best practices/what you would recommend? I'll be running an NGINX proxy in front of mlflow by default, but that's just meant to handle authentication.

**Comment by aarondav** [comment]
Great question - the main limitation of the current approach is that the flask app is single-threaded, so it can only handle one request at a time. Since the server is basically just relaying local files to the user, requests are extremely short, so this isn't shouldn't be a huge limitation for the usage pattern we would expect initially.

It would probably be a good idea to have a [pre-packaged WSGI](http://flask.pocoo.org/docs/0.12/deploying/wsgi-standalone/) within the mlflow ui for handling multiple requests, so people don't have to figure out how to configure and install a separate WSGI server in front of it -- no one is currently working on that, but it shouldn't be hard to add.

**Comment by zegerhoogeboom** [comment]
Thanks for the response, indeed this isn't a limitation for my intended usage. I tried for a little bit to install a WSGI server in front of the UI but gave up as it seemed to be more involved than I was willing to spend time on it. I opened the issue in the first place as running the UI in a docker container didn't work as the interface was binding to `127.0.0.1` and not `0.0.0.0`, but I believe this is already resolved through other issues (did the same thing on my own fork). Running in a docker container is required for me as I'm running a NGINX proxy in front of all my public facing docker containers.

The UI is currently running fine but I now have some other problems as the UI is running on HTTPS. I'll try to figure that out soon though. Feel free to close this issue if you feel it's irrelevant but it's maybe good to keep it open also for the HTTPS remote server support.

**Comment by waltchan86** [comment]
could i ask how we change the UI binding from 127.0.0.1 to 0.0.0.0? Thanks

**Comment by zegerhoogeboom** [comment]
Actually the default is now 0.0.0.0 on the current `master`. See https://github.com/databricks/mlflow/blob/9e7ec0093a72f4cbe3d55629c7e2cc5043068e12/mlflow/cli.py#L132

**Comment by brookesey** [comment]
Is there a way to change to port number the UI runs on?

**Comment by mateiz** [comment]
FYI this should be fixed in the master branch by https://github.com/databricks/mlflow/pull/61. It uses gunicorn as a WSGI server and it also lets you control the host and port to bind on.

**Comment by Unliklll** [comment]
For this problem，is there effective solutions available now？

**Comment by mateiz** [comment]
Yes, the gunicorn based server was released in MLflow 0.2. Take a look at https://mlflow.org/docs/latest/tracking.html#running-a-tracking-server for a guide on how to run it.

**Comment by Unliklll** [comment]
> Yes, the gunicorn based server was released in MLflow 0.2. Take a look at https://mlflow.org/docs/latest/tracking.html#running-a-tracking-server for a guide on how to run it.

when Serving the Model
1) mlflow sklearn serve /Users/mlflow/mlflow/mlruns/<experiment_id>/<run_id>/artifacts/model -p 1234
2)mlflow server  --file-store /mnt/persistent-disk --default-artifact-root s3://my-mlflow-bucket/ --host 0.0.0.0
It is clear that How are the parameters configured in 1).   Can you make it more clearer by using a real example in 2) ?  That is to say how will "/mnt/persistent-disk" and "s3://my-mlflow-bucket/" be configured in a real example?  Thank you very much.

**Comment by aarondav** [comment]
Please check the following section, "Storage" -- that discusses those exact two parameters :)
